src/helpers/ws.ts

```diff
@@ -126,30 +122,84 @@ export class WebsocketService {
   /**
    * This function starts the websocket subscription with the websocket provider
    *
-   * @param socketUrl - the websocket URL to use
-   * @param appId - the App ID
-   *
    * @returns a promise that resolves when the websocket connection is established
    */
-  init(socketUrl: string, appId: string): Promise<unknown> {
-    this.websocketUrl = socketUrl;
-    this.appId = appId;
+  init(): Promise<unknown> {
+    this.connect();
 
-    this.globalSocket = new WebSocket(this.websocketUrl);
+    return this.initPromise.promise;
+  }
 
-    this.globalSocket.onerror = this.handleError;
-    this.globalSocket.onclose = this.handleClosed;
-    this.globalSocket.onopen = this.handleOpen;
-    this.globalSocket.onmessage = this.handleMessage;
+  private connect() {
+    this.getWebSocketConnection(33024, 33029)
+      .then((webSocket) => {
+        this.globalSocket = webSocket;
+        this.globalSocket.onerror = this.handleError;
+        this.globalSocket.onclose = this.handleClosed;
+        this.globalSocket.onopen = this.handleOpen;
+        this.globalSocket.onmessage = this.handleMessage;
+
+        this.handleOpen();
+      }).catch(() => {
+        this.reconnect();
+      });
+  }
 
-    return this.initPromise.promise;
+  private reconnect() {
+    setTimeout(() => {
+      this.globalSocket.close();
+      this.connect();
+    }, 2000);
+  }
+
+  private getWebSocketConnection(startPort: number, endPort: number): Promise<WebSocket> {
+    const webSocketUrl = getWebsocketUrl(asperaDesktop.globals.desktopUrl);
+
+    const checkPort = (port: number): Promise<WebSocket> => {
+      return new Promise((resolve, reject) => {
+        const webSocket = new WebSocket(`${webSocketUrl}:${port}`);
+
+        webSocket.onopen = () => {
+          resolve(webSocket);
+        };
+
+        webSocket.onerror = () => {
+          reject(`Connection failed on port ${port}`);
+        };
+      });
+    };
+
+    return new Promise((resolve, reject) => {
+      let currentPort = startPort;
+
+      const connectPort = () => {
+        if (currentPort > endPort) {
+          return reject('No available WebSocket connection found');
+        }
+
+        checkPort(currentPort)
+          .then(ws => resolve(ws))
+          .catch(() => {
+            currentPort++;
+            connectPort();
+          });
+      };
+
+      connectPort();
+    });
   }
 
   private notifyEvent(event: WebsocketEvents) {
     if (typeof this.eventListener === 'function') {
       this.eventListener(event);
     }
   }
+
+  private updateRPCPort() {
```

Comment:
Nitpick, but `updateRpcPort` is more readable IMO.

Reply:
Done!